client/src/GardenPlants/GardenPlantForm.js

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import Api from "../Api";
+
+function GardenPlantForm() {
+    const { id } = useParams();
+    const history = useHistory();
+    const [gardens, setGardens] = useState([]);
+    const [plants, setPlants] = useState([]);
+    const [gardenplant, setGardenPlant] = useState({
+        GardenId: '',
+        PlantId: '',
+    });
+
+    useEffect(function () {
+        Promise.all([Api.gardens.index(), Api.plants.index()]).then((responses) => {
```

Comment:
Here, we fetch the list of gardens and of plants, then set those lists into the state variables above.

Then, we fetch the gardenplant we're editing (if we're editing), or set the initial garden and plant ids into the gardenplant (if we're creating).